### PR TITLE
Merchandising Page

### PR DESCRIPTION
**Pull request by bsuravech**
**Summary**
[MAG-109] Merchandising admin page to showcase all merchandising products.

**Result**
- New admin page replacing the Landing Page Builder link with Merchandising
- Plan based access links for all products using the proxy
- Created an upsell-plan.phtml for a resuable block template in UI 

This is dependent on Query Merchandiser. 

[MAG-109]: https://algolia.atlassian.net/browse/MAG-109

### CODE REVIEW

File: Controller/Adminhtml/Merchandising/Index.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\View\LayoutFactory;
+
+class Index extends \Magento\Backend\App\Action
+{
+    /** @var ResultFactory */
+    protected $resultFactory;
+
+    /** @var LayoutFactory */
+    protected $layoutFactory;
```
Comment (damcou) [inline Controller/Adminhtml/Merchandising/Index.php]: Is it useful?

Comment (bsuravech) [reply Controller/Adminhtml/Merchandising/Index.php]: Good catch! I'll update it!

File: ViewModel/Adminhtml/Merchandising/Page.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    /**
+     * @return bool
+     */
+    public function canAccessLandingPageBuilder()
+    {
+        $planLevel = 1;
+
+        $planLevelInfo = $this->proxyHelper->getInfo(ProxyHelper::INFO_TYPE_PLAN_LEVEL);
```
Comment (JanPetr) [inline ViewModel/Adminhtml/Merchandising/Page.php]: `planLevel` is included in `ALL` and can be get from `ProxyHelper:: getClientConfigurationData()` as well. Could it be refactored and do only one request to Proxy? :)

Comment (bsuravech) [reply ViewModel/Adminhtml/Merchandising/Page.php]: Oh amazing. Can I update @damcou 's  LPB to do the same? Or do you want that in a separate PR?

File: ViewModel/Adminhtml/Merchandising/Page.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+namespace Algolia\AlgoliaSearch\ViewModel\Adminhtml\Merchandising;
+
+use Algolia\AlgoliaSearch\Helper\ProxyHelper;
+
+class Page
+{
+    /** @var ProxyHelper */
+    private $proxyHelper;
+
+    public function __construct(
```
[the remark this answers is not in the document]
Comment (bsuravech) [reply ViewModel/Adminhtml/Merchandising/Page.php]: Yes we can!

File: ViewModel/Adminhtml/Merchandising/Page.php (hunk 2 above)
Comment (JanPetr) [reply ViewModel/Adminhtml/Merchandising/Page.php]: Feel free to update it here :)

**Pull request merged**